querybook/server/models/data_element.py

```diff
@@ -0,0 +1,83 @@
+import sqlalchemy as sql
+from app import db
+from const.db import mediumtext_length, name_length, now, type_length
+from const.metastore import DataElementAssociationProperty, DataElementAssociationType
+from lib.sqlalchemy import CRUDMixin
+from sqlalchemy.orm import relationship
+
+Base = db.Base
+
+
+class DataElement(CRUDMixin, Base):
+    __tablename__ = "data_element"
+
+    id = sql.Column(sql.Integer, primary_key=True)
+    created_at = sql.Column(sql.DateTime, default=now)
+    updated_at = sql.Column(sql.DateTime, default=now)
+    created_by = sql.Column(
+        sql.Integer, sql.ForeignKey("user.id", ondelete="SET NULL"), nullable=True
+    )
+    metastore_id = sql.Column(
+        sql.Integer,
+        sql.ForeignKey("query_metastore.id", ondelete="CASCADE"),
+        nullable=False,
+    )
+    name = sql.Column(
+        sql.String(length=name_length), index=True, unique=True, nullable=False
+    )
+    # The string representation of the data element’s underlying data type
+    type = sql.Column(sql.String(length=type_length), nullable=False)
+    description = sql.Column(sql.Text(length=mediumtext_length), nullable=True)
+    properties = sql.Column(sql.JSON, nullable=True)
+
+    def to_dict(self):
+        return {
+            "id": self.id,
+            "metastore_id": self.metastore_id,
+            "name": self.name,
+            "type": self.type,
+            "description": self.description,
+            "properties": self.properties,
+        }
+
+
+class DataElementAssociation(CRUDMixin, Base):
+    __tablename__ = "data_element_association"
+
+    id = sql.Column(sql.Integer, primary_key=True)
+    created_at = sql.Column(sql.DateTime, default=now)
+    updated_at = sql.Column(sql.DateTime, default=now)
+    column_id = sql.Column(
+        sql.Integer,
+        sql.ForeignKey("data_table_column.id", ondelete="CASCADE"),
+        nullable=False,
+    )
+    type = sql.Column(
+        sql.Enum(DataElementAssociationType),
+        default=DataElementAssociationType.REF,
+        nullable=False,
+    )
+    property_name = sql.Column(
```

Comment:
make this strings? otherwise we are gonna have fun implementing rows in future if needed lol